### PR TITLE
feat: Add data validation for subscription

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:
* Add data validation for subscription
* Add method for get allow list
* Test run:
    * Deployed with `--enableSubscriptions true` and verified validations done on Subscription resources
    * Deployed without `--enableSubscriptions` option and verified validations are not done on Subscription resources

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: USING_SUBSCRIPTIONS.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+## Enabling multi-tenancy
+
+Subscriptions are an opt-in feature. Enabling subscriptions incurs a cost even if there are no active subscriptions. 
+
+To enable it, follow the steps below:
+  
+1. Update the method `getAllowListedSubscriptionEndpoints` in `src/subscriptions/allowList.ts` to include endpoints allowed for recieving subscriptions.
+
+2. Use the `enableSubscriptions` option and deploy the stack
```
Comment (carvantes) [inline USING_SUBSCRIPTIONS.md]:
```suggestion
2. Use the `enableSubscriptions` option when deploying the stack
```

File: USING_SUBSCRIPTIONS.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Supported Subscriptions
+
+Subscriptions satisfying the following restrictions are supported:
+
+1. channel.type must be rest-hook 
+2. channel.endpoint must be allow-listed
+3. channel.endpoint must use HTTPS
+4. channel.payload must be application/fhir+json if present.
+5. channel.criteria must be a valid search query that is supported by FWoA. This is covered in detail in the next sections.
```
Comment (carvantes) [inline USING_SUBSCRIPTIONS.md]:
```suggestion
5. channel.criteria must be a valid search query that is supported by FWoA.
```

File: src/config.ts [hunk 3]
```diff
@@ -65,72 +67,81 @@ const esSearch = new ElasticSearchService(
...
-    auth: {
-        authorization: authService,
-        // Used in Capability Statement Generation only
-        strategy: {
-            service: 'OAuth',
-            oauthPolicy: {
-                authorizationEndpoint: `${OAuthUrl}/authorize`,
-                tokenEndpoint: `${OAuthUrl}/token`,
+export const getFhirConfig = async (): Promise<FhirConfig> => {
+    if (process.env.ENABLE_SUBSCRIPTIONS) {
```
Comment (carvantes) [inline src/config.ts]: This won't work as expected. env vars are always strings, and the string `'false'` is truthy

**Review by carvantes — CHANGES_REQUESTED** [review]

File: src/config.ts (hunk 3 above)
Comment (Bingjiling) [reply src/config.ts]: Ah! Good catch! Will update.

**Review by carvantes — APPROVED** [review]

**Pull request merged**